### PR TITLE
Update from git

### PR DESCRIPTION
**Pull request by Zakarialabib**
Hey, in this pullrequest, i tried to make updating the project more easier, myStockMaster user can recieve updates from stable branch.

### CODE REVIEW

File: app/Http/Livewire/Settings/Update.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ];
+
+    public $message = "";
+
+    public function updateProject()
+    {
+      
+        $gitHandler = new GitHandler();
+        $updated = $gitHandler->fetchAndPull();
+        // $updatedPush = $gitHandler->pushChanges();
```
Comment (Daveyvdweide) [inline app/Http/Livewire/Settings/Update.php:22]:
```suggestion
```

File: resources/views/livewire/settings/update.blade.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            <div class="text-gray-600">
+                    {{ $message }}
+                </div>
+            @endif
+            <x-button type="" x-on:click="update = false" primary>
+            {{('Close')}}
+            </x-button>
+        </div>
+    </div>
+</div>
```
Comment (Daveyvdweide) [inline resources/views/livewire/settings/update.blade.php:23]: Missing new line

File: app/Http/Livewire/Settings/Update.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        $this->message = 'Update your  system with the latest version';
+    }
+
+    public function optimize()
+    {
+        //  Artisan::call('optimize:clear');
+        // Artisan::call('optimize');
+        // exec("composer dump");
+    }
```
Comment (Daveyvdweide) [inline app/Http/Livewire/Settings/Update.php:44]:
```suggestion
```

Remove unnecessary functions to declutter code base

File: app/Http/Livewire/Settings/Update.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            return "updating project";
+    } else {
+            return "Error updating project";
+        }
+
+    }
+
+    public function mount()
+    {
+        $this->message = 'Update your  system with the latest version';
```
Comment (Daveyvdweide) [inline app/Http/Livewire/Settings/Update.php:36]:
```suggestion
        $this->message = 'Upgrade to latest version';
```

Grammar improvement

File: app/Http/Livewire/Settings/Update.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        // $updatedPush = $gitHandler->pushChanges();
+    
+        if ($updated) {
+            return "Project featched";
+        } elseif ($updatedPush){
+            return "updating project";
+    } else {
+            return "Error updating project";
+        }
+
```
Comment (Daveyvdweide) [inline app/Http/Livewire/Settings/Update.php:31]:
```suggestion
        if ($updated) {
            return 'Project fetched';
        } elseif ($updatedPush){
            return 'updating project';
        } 
        
        return 'Error updating project';
```
1. Use `'string'` for string.
2. Fixed a spelling error.
3. Removed unnecessary `else`

File: app/Helpers/GitHandler.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+            // Push the changes to the remote repository
+            exec("git push $repoUrl $repoBranch", $output, $return_var);
+
+            if ($return_var === 0) {
+                $this->message = "Changes pushed successfully.";
+            } else {
+                $this->message = "Error pushing changes.";
+            }
+        }
```
Comment (Daveyvdweide) [inline app/Helpers/GitHandler.php:58]:
```suggestion
        exec("git diff --exit-code", $diff, $return);

        if ($return === 0) {
            $this->message = "No changes to push.";
            return;
        } 
        // Get the repository URL and branch from the .env file
        $repoUrl = env('REPO_URL');
        $repoBranch = env('REPO_BRANCH');

        // Push the changes to the remote repository
        exec("git push $repoUrl $repoBranch", $output, $return);

        if ($return=== 0) {
            $this->message = "Changes pushed successfully.";
            return;
        } 
            
        $this->message = "Error pushing changes.";
    }
```

Try to keep nesting `if-statements` to a bare minimum

File: app/Helpers/GitHandler.php [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                    $this->message = "Error cleaning untracked files and directories.";
+                }
+            } else {
+                $this->message = "Error resetting to latest updates from origin/$branch.";
+            }
+        } else {
+            $this->message = "Error fetching updates from origin/$branch.";
+        }
+
+        return $this->message;
```
Comment (Daveyvdweide) [inline app/Helpers/GitHandler.php:35]: this needs some refactoring

**Review by Daveyvdweide — CHANGES_REQUESTED** [review]

File: app/Helpers/GitHandler.php (hunk 6 above)
Comment (Zakarialabib) [reply app/Helpers/GitHandler.php:58]: I will keep that in mind

File: app/Helpers/GitHandler.php (hunk 7 above)
Comment (Zakarialabib) [reply app/Helpers/GitHandler.php:35]: Thank you for contributions🙏🎉

Comment (Daveyvdweide) [reply app/Helpers/GitHandler.php]: @Zakarialabib np!

I'd start of by creating an abstraction layer around the command you want to call

**Pull request merged**